lib/internal/modules/esm/module_job.js

```diff
@@ -93,6 +97,28 @@ class ModuleJob {
       }
     } catch (e) {
       decorateErrorStack(e);
+      if (StringPrototypeIncludes(e.message,
+                                  ' does not provide an export named')) {
+        const [ parentFileUrl, importStatement ] =
```

Comment:
note that using array destructuring here means that `delete Array.prototype[Symbol.iterator]` will break this code. it'd be safer to use explicit index property access (but this is a big problem in node core overall, so nbd if you want to leave it here)